src/services/classifier.ts

```diff
@@ -152,8 +152,8 @@ export function createClassifier(): Classifier {
                 handleToken();
                 lastNonTriviaToken = token;
             }
-            const end = scanner.getTextPos();
-            pushEncodedClassification(scanner.getTokenPos(), end, offset, classFromKind(token), spans);
+            const end = scanner.getTokenEnd();
+            pushEncodedClassification(scanner.getTokenStart(), end, offset, classFromKind(token), spans);

```

Comment:
we have some weird patterns with saving getTokenStart/getTokenEnd into temp variables, even though those functions are stateless.

Reply:
It's inefficient to call scanner methods over and over again needlessly ;)

Reply:
I hate to suggest it but... what if we destructured these for perf too? 😄

Reply:
Oh, this is classifier. Probably pointless.

Reply:
I was thinking about this recently - it probably makes sense in the parser itself.